### ISSUE TITLE
gci auto fix is broken

### ISSUE THREAD

**Issue by Dreamacro** [golangci/golangci-lint#2591]
### Welcome

- [X] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've included all information below (version, config, etc).
- [X] Yes, I've tried with the standalone linter if available. (https://golangci-lint.run/usage/linters/)

### Description of the problem

gci only report issues but it won't fix it

### Version of golangci-lint

<details>

```console
$ golangci-lint --version
golangci-lint has version v1.44.1
```

</details>

### Configuration file

<details>

```console
$ cat .golangci.yml
linters:
  disable-all: true
  enable:
    - gofumpt
    - govet
    - gci

issues:
  fix: true

linters-settings:
  gci:
    sections:
      - standard
      - prefix(github.com/pkg/pkg)
      - default

```

</details>

### Go environment

<details>

```console
$ go version && go env
go version go1.17.6 darwin/arm64
```

</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean
$ golangci-lint run -v
INFO [runner] linters took 924.918333ms with stages: goanalysis_metalinter: 924.218458ms
INFO fixer took 0s with no stages
component/dialer/bind_darwin.go:7:4: Expected 'i', Found 'o' at component/dialer/bind_darwin.go[line 7,col 4] (gci)
	"golang.org/x/sys/unix"
	  ^
INFO File cache stats: 1 entries of total size 1.4KiB
INFO Memory: 13 samples, avg is 273.7MB, max is 352.6MB
INFO Execution took 1.127992166s

```

</details>

### Code example or link to a public repository

<details>

```go
import (
	"net"
	"syscall"

	"golang.org/x/sys/unix"

	"github.com/pkg/pkg"
)
```

</details>

**Comment by ldez** [comment]
Hello,

since the last gci update (#2532), autofix is no longer supported.

**Comment by SVilgelm** [comment]
Yeah, this was the main feature of the gci, now I have to revert golangci-linter to v1.44.0

**Comment by SVilgelm** [comment]
there are a lot of questions to gci, the auto fix is just one of them. The error messages are completely unclear, like `expected \t, but got \n`. I will open issues on gci repo.

About restoring, I see that gci can do it if we use gci cli, probably we should talk with the @daixiang0

**Comment by SVilgelm** [comment]
@ldez I see that the logic to prepare a diff was removed in that PR
```go
err = differ.Diff(&diff, bytes.NewReader(source), bytes.NewReader(result))
if err != nil {
	return nil, fmt.Errorf("can't get gci diff output: %v", err)
}
```

I remember I had to implement it because the `gci` did not support analyzer at that time. Can we restore it?

**Comment by ldez** [comment]
> Can we restore it?

no because gci is now an analyzer, we don't have like goimports/gofmt a "diffable" outout
